docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Machine ID Architecture
+description: How Teleport Machine ID works.
+---
+
+This section provides an overview of Teleport Machine ID's inner workings.
+
+A more in-depth specification of the workings of Machine ID can be found in
+[the Request For Discussion.](https://github.com/gravitational/teleport/blob/master/rfd/0064-bot-for-cert-renewals.md)
+
+## Bot creation
+
+When creating a bot via `tctl`, up to three resources are created:
+
+- Bot user: this will be the user that the Machine ID agent authenticates as.
+- Bot role: the bot user is assigned the bot role, and the bot role contains
+  various permissions that the bot will need to function. For example, the
+  ability to watch the certificate authorities and the ability to 
+  [impersonate roles](#role-impersonation).
+- Token: for [onboarding](#onboarding), a token must exist that allows the
+  Machine ID agent to initially authenticate as the bot user. If an existing
+  token is not specified, then a static single-use token will be created by the
+  Auth Server.
+
+## `tbot`
+
+`tbot` is the binary that acts as the agent for Machine ID.
+
+On initial load, `tbot` uses the [configured join method](#onboarding) to obtain
+a set of credentials for the bot user from the Teleport Auth Service. It can 
+then use these credentials to communicate with the Teleport Auth Service as the 
+bot.
+
+On a configured regular period `tbot` then:
+
+1. Renews these bot user credentials
+2. For each destination provided in `tbot`'s configuration:
```

Comment:
I think we should rephrase this sentence to assume that a reader doesn't know yet what a "destination" is, since it is the first mention of the term on this page.